### ISSUE TITLE
Cannot delete physically absent network card entry using WebUI

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-core#2775]
Thanks to forum user [atg2004](https://forum.rockstor.com/u/atg2004), an issue has been highlighted, where, in case of a network card being removed/replaced (in this case the entire motherboard was replaced), the orphaned network connection cannot be removed using the WebUI, likely due to the reported uuid/interface not existing anymore.

Forum threads for reference:

https://forum.rockstor.com/t/how-to-remove-lan-after-replaced-mobo-board/9196

and continued with specific error message here:

https://forum.rockstor.com/t/not-supported-between-instances-of-nonetype-and-int/9198

In this case of non-existence, Rockstor should probably still delete the "phantom" entry (e.g. with an informational message as part of the current deletion confirmation pop-up) when the user selects the `delete` (trashcan) button.

The traceback reported is this:
```
File “/opt/rockstor/src/rockstor/rest_framework_custom/generic_view.py”, line 41, in _handle_exception
yield
File “/opt/rockstor/src/rockstor/storageadmin/views/network.py”, line 548, in delete
if nco.bridgeconnection_set.first() > 0: # If docker network
TypeError: ‘>’ not supported between instances of ‘NoneType’ and ‘int’
```

So the corresponding code (testing branch) can be found here (see line 548):

https://github.com/rockstor/rockstor-core/blob/b870fa2f0264a8696440cccbc2526c50617ef773/src/rockstor/storageadmin/views/network.py#L545-L556

**Comment by phillxnet** [comment]
@Hooverdan96 I'm attempting to create a reproducer for this issue, to test if linked PR:
"TypeError when deleting unused Rocknet #2814" #2817
also fixes this same issue (strongly suspected).

However when creating a VM instance with 2 ethernet interfaces to receive the follows:

![2-network-devices](https://github.com/rockstor/rockstor-core/assets/2521585/ed88738f-7b51-4c9c-b19f-bd34a0dc374b)

And then shutting this VM down, and removing the last added network device. The next power up gives the expected auto-correction of only the existing devices showing within the Web-UI:

![remaining-network-device](https://github.com/rockstor/rockstor-core/assets/2521585/19045a11-a763-48b6-aea9-4315c0a9f87a)

Is there something I'm missing here. The test system does not yet have the proposed fix associated with the Rocknet issue.

**Comment by phillxnet** [comment]
@Hooverdan96 & @FroggyFlox OK, found a reproducer: tangential to original but should serve:

- VM instance as in the last comment: where 2 interfaces are active and identified (first pic).
- Delete one via bin icon:

![delete-existing-interface-None-and-int-error](https://github.com/rockstor/rockstor-core/assets/2521585/e5f98a17-784f-42b8-b44a-2fb8370cc98a)

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2817
